cmd/agent/app/integrations.go

```diff
@@ -257,19 +315,17 @@ func pip(args []string, stdout io.Writer, stderr io.Writer) error {
 }
 
 func install(cmd *cobra.Command, args []string) error {
+	if err := detectEnvironment(); err != nil {
```

Comment:
Because we need this for every subcommand, shouldn't we move the detect phase into the init section ?

Reply:
I wanted to as well, but we don't want to do too much fs stuff in there so was asked not to